client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -368,6 +395,53 @@ private static String getChildElementString(Element parent, String tagName) {
         return null;
     }
 
+    /**
+     * Returns a hash that should be consistent for any individual swarm client (as long as it has a persistent IP)
+     * and should be unique to that client.
+     *
+     * @param remoteFsRoot the file system root should be part of the hash (to support multiple swarm clients from 
+     *                     the same machine)
+     * @return our best effort at a consistent hash
+     */
+    public static String hash(File remoteFsRoot) {
+        StringBuilder buf = new StringBuilder();
+        try {
+            buf.append(remoteFsRoot.getCanonicalPath()).append('\n');
+        } catch (IOException e) {
+            buf.append(remoteFsRoot.getAbsolutePath()).append('\n');
+        }
+        try {
+            for (NetworkInterface ni : Collections.list(NetworkInterface.getNetworkInterfaces())) {
+                for (InetAddress ia : Collections.list(ni.getInetAddresses())) {
+                    if (ia instanceof Inet4Address) {
+                        buf.append(ia.getHostAddress()).append('\n');
+                    }
+                }
+                byte[] hardwareAddress = ni.getHardwareAddress();
+                if (hardwareAddress != null) {
+                    buf.append(Arrays.toString(hardwareAddress));
+                }
+            }
+        } catch (SocketException e) {
+            // oh well we tried
+        }
+        return getDigestOf(buf.toString()).substring(0, 8);
+    }
+
+    public static String getDigestOf(String text) {
+        try {
```

Comment:
Apache commons has some nice util classes (```org.apache.commons.codec.binary.Hex``` and
```org.apache.commons.codec.digest.DigestUtils```) that would make this a bit less bloated, and eliminate the need for ```toHexString```